### PR TITLE
Added support for auto-replanting of Sugar Cane, Pumpkin Stems, and Melon Stems

### PR DESCRIPTION
**Pull request by thehale**
Also did a little bit for code formatting for easier reading.

### CODE REVIEW

**Comment by Geolykt** [comment]
I might be fully wrong there, but I recall the case blocks being 'unindented' by one layer. The switch-case Statements have many oddities (mainly because of fall-through), and I believe that's one of them.

The reason behind that (If I recall correctly), is that the case instructions are basically just jump instructions, only somewhat in reverse. Due to fall-through, the whole Switch block is a single block, however the case Statements are unindented by one layer for visibillity

**Comment by thehale** [comment]
@Geolykt It's honestly a style preference. This is Java and not Python so we can indent things however we like.

You are right that switch statements are a special form of jump where the target address can be calculated at runtime without having to check each case (that's why they are so much faster than if-elseif-else when you have lots of cases). Fall through only occurs because the compiler doesn't automatically place another jump instruction at the end of each case. That's why you have to make sure to always add a break or return unless you actually want fall through (which does have its uses).

All the technicalities aside, there's nothing in Java that requires a certain indentation. I could use a different indentation for each case and the Java compiler wouldn't care.

I personally prefer indenting the cases because the code is easier for me to read that way. I see the keyword "switch" as encapsulating its various cases just as a for loop encapsulates the code it iterates. 

If @toomanybugs1 would prefer to have them unindented he can request revisions and I can update the pull request.

TL;DR Java lets us indent switch-case however we like, and this pull request reflects my personal preference. I can resubmit if requested.

**Comment by toomanybugs1** [comment]
Hey thanks for another contribution! I hadn't even thought of adding those crops. I just merged a pull request so there is a small conflict in Main.java. I may just go and fix that on my side because it's pretty small. As far as the switches go, I personally prefer the indentation for cases.

**Comment by toomanybugs1** [comment]
Okay so I was testing your branch and I get an occasional error when harvesting. It looks like its being triggered at 120. It's trying to drop air because for some reason it thinks that's what you're harvesting. 

`[16:05:25 ERROR]: Could not pass event BlockBreakEvent to QuickFarm v1.1
java.lang.IllegalArgumentException: Cannot drop air
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.EntityItem.setItemStack(EntityItem.java:443) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.EntityItem.<init>(EntityItem.java:42) ~[patched_1.16.1.jar:git-Paper-114]
        at org.bukkit.craftbukkit.v1_16_R1.CraftWorld.dropItem(CraftWorld.java:736) ~[patched_1.16.1.jar:git-Paper-114]
        at org.bukkit.craftbukkit.v1_16_R1.CraftWorld.dropItemNaturally(CraftWorld.java:751) ~[patched_1.16.1.jar:git-Paper-114]
        at io.github.toomanybugs1.QuickFarm.Main.tryAutoReplant(Main.java:120) ~[?:?]
        at io.github.toomanybugs1.QuickFarm.Main.onBlockBreak(Main.java:89) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor502.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.1.jar:git-Paper-114]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.1.jar:git-Paper-114]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.1.jar:git-Paper-114]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:607) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.PlayerInteractManager.breakBlock(PlayerInteractManager.java:324) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.PlayerInteractManager.a(PlayerInteractManager.java:283) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.PlayerInteractManager.a(PlayerInteractManager.java:173) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.PlayerConnection.a(PlayerConnection.java:1389) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.PacketPlayInBlockDig.a(SourceFile:40) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.PacketPlayInBlockDig.a(SourceFile:10) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:23) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.TickTask.run(SourceFile:18) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.MinecraftServer.aZ(MinecraftServer.java:1136) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.MinecraftServer.executeNext(MinecraftServer.java:1129) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.MinecraftServer.sleepForTick(MinecraftServer.java:1090) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.MinecraftServer.v(MinecraftServer.java:1004) ~[patched_1.16.1.jar:git-Paper-114]
        at net.minecraft.server.v1_16_R1.MinecraftServer.lambda$a$0(MinecraftServer.java:177) ~[patched_1.16.1.jar:git-Paper-114]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_251]`

**Comment by thehale** [comment]
Oh! That's an exciting error I haven't seen. Let me dig into it to see if I can figure out what's going on.

**Comment by toomanybugs1** [comment]
If it helps it happened only when I was trying to break pumpkin/melon stems!

**Comment by thehale** [comment]
@toomanybugs1 I did some more testing this weekend and only came across the error when breaking melon/pumpkin stems that were in the first stage of growth. That said, even though the error was thrown, it didn't mess with the auto-replant.

To me, it seems like this air drop error is the result of the high probability of a early stage melon/pumpkin stem not dropping anything. I added a simple check to make sure we weren't attempting to drop air and that resolved it on my end. Feel free to double check me and see what your experience is.

Lastly, I've discovered that the auto-replant only works for un-attached melon/pumpkin stems (even though I've included the attached version in the code). I'll keep trying to figure that out, but it's minor enough that you can still merge in the changes so people can start benefitting from the new crops.

**Comment by toomanybugs1** [comment]
That's great, I think part of it was that I was in creative mode, which causes it not to drop. I will look into the non-attached problem as well. For now I think this is perfect. Thanks for the contribution!
